src/cloud/rollout.js

```diff
@@ -0,0 +1,94 @@
+import kleur from 'kleur';
+import { exec, execSyncInherit } from '../util/shell';
+import fs from 'fs';
+import path from 'path';
+
+export function getDeployment(service, subservice) {
+  let deployment = service;
+  if (subservice) deployment += `-${subservice}`;
+  deployment += '-deployment';
+  return deployment;
+}
+
+async function getTag() {
+  let tag = await exec('git tag --points-at HEAD');
+  if (!tag) tag = await exec('git rev-parse --short --verify HEAD');
+  const stat = await exec('git diff --stat');
+  if (stat) tag += '-dirty';
+  return tag;
+}
+
+async function getMetaData() {
+  const author = await exec('git config user.name');
+  const date = new Date().toUTCString();
+  const branch = await exec('git branch --show-current');
+  const git = await getTag();
+  let fields = `\\"author\\":\\"${author}\\"`;
```

Comment:
could this not just be `JSON.stringify`?

Reply:
That's what I started out with, but unfortunately no. This relates to the issue with `exec` and double quotes.

Reply:
ahh of course I get it now..... but maybe just `JSON.stringify(...).replace(/"/g, "\\")`? maybe exec could even do it?

Reply:
Fixed